### PR TITLE
Upgrading notes for 2.7

### PR DESCRIPTION
**Pull request by rgugliel-da**

### CODE REVIEW

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst [hunk 1]
```diff
@@ -180,6 +180,99 @@ Finally, you can ping the participant to see if the system is operational
...
+This mode can be enabled by setting the appropriate storage parameter:
+
+.. code:: bash
+
+    canton.X.Y.storage.parameters.migrate-and-start = yes
+
+In order to benefit from the new security features brought by protocol version 5,
+you will have to :ref:`upgrade the domain accordingly <canton_domain_protocol_version_upgrading>`.
+
+Activation of unsupported features
```
Comment (dasormeter) [inline docs/2.7.0/docs/canton/usermanual/upgrading.rst:198]: @rgugliel-da is there a specific unsupported feature it is recommended to activate, or is this just a general note on the change in mechanics?

Comment (rgugliel-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:198]: The security features there: https://github.com/DACH-NY/canton/blob/6357b20b7125777ca0713c7248df59684a217c86/UNRELEASED.md#additional-security-validations

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst [hunk 2]
```diff
@@ -180,6 +180,99 @@ Finally, you can ping the participant to see if the system is operational
 Version Specific Notes
 ~~~~~~~~~~~~~~~~~~~~~~
 
+Upgrade to Release 2.7
+^^^^^^^^^^^^^^^^^^^^^^
+Version 2.7 will slightly extend the database schema used. Therefore, you will have to perform the
+database migration steps.
```
Comment (soren-da) [inline docs/2.7.0/docs/canton/usermanual/upgrading.rst:186]: add a link to the migration steps

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 1 above)
Comment (soren-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:198]: I don't understand your comment Raf, the security enhancements are not opt-in, only a few are opt-out which caused problems with CN, but I wouldn't recomment that to clients to disable.

I think Brian's question still stands which unsupported features require the dev version support to be enabled.

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst [hunk 3]
```diff
@@ -180,6 +180,99 @@ Finally, you can ping the participant to see if the system is operational
...
+    )
+
+
+Unique Contract Keys Deprecation
+""""""""""""""""""""""""""""""""
+The unique-contract-keys parameters for both participant and domain nodes are now marked as deprecated.
+As of this release, the meaning and default value (true) remain unchanged.
+However, contract key uniqueness will not be available in the next major version, featuring multi-domain connectivity.
+If you are already setting this key to false explicitly (preview), this behavior will be the default one after the configuration key is removed.
+If you don't explicitly set this value to false, you are encouraged to evaluate evolving your existing applications and services to avoid relying on this feature.
```
Comment (soren-da) [inline docs/2.7.0/docs/canton/usermanual/upgrading.rst:230]: are we printing to the logs a warning/info if it is still true?

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst [hunk 4]
```diff
@@ -180,6 +180,99 @@ Finally, you can ping the participant to see if the system is operational
...
+
+The error codes and metadata of GRPC errors returned as part of failed command interpretation from the Ledger API have been updated to include more information.
+Previously, most errors from the Daml engine would be given as either `GenericInterpretationError` or `InvalidArgumentInterpretationError`.
+They now all have their own codes and encode relevant information in the GRPC Status metadata.
+Specific error changes are as follows:
+* `GenericInterpretationError` (Code: `DAML_INTERPRETATION_ERROR`) with GRPC status `FAILED_PRECONDITION` is now split into:
+
+    * `DisclosedContractKeyHashingError` (Code: `DISCLOSED_CONTRACT_KEY_HASHING_ERROR`) with GRPC status `FAILED_PRECONDITION`
+    * `UnhandledException` (Code: `UNHANDLED_EXCEPTION`) with GRPC status `FAILED_PRECONDITION`
+    * `UserError` (Code: `USER_ERROR`) with GRPC status `FAILED_PRECONDITION`
```
Comment (soren-da) [inline docs/2.7.0/docs/canton/usermanual/upgrading.rst:259]: Unrelated: USER_ERROR and DEV_ERROR are unfortunate generic error codes for something that daml engine specific

**Review by soren-da — APPROVED** [review]

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 1 above)
Comment (daravep) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:198]: Not a big issue, but I wouldn't add "how to enable unsupported features" as part of the "how to migrate production systems" guide.

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 4 above)
Comment (daravep) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:259]: @samuel-williams-da is changing the codes.

**Review by daravep — APPROVED** [review]
LGTM, though we might have to change the ledger api error codes when samuel changes them.

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 4 above)
Comment (samuel-williams-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:259]: I will update them to `InterpretationUserError` and `InterpretationDevError` with their snake cases respectively.

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 1 above)
Comment (rgugliel-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:198]: Will drop.

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 3 above)
Comment (rgugliel-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:230]: I don't think so. @stefanobaghino-da do you know if a warning is printed ?

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 4 above)
Comment (rgugliel-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:259]: Yep, I reviewed his PR, will push a commit here

Comment (samuel-williams-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:259]: The generator for these docs is on the daml repo, and has been updated in this PR
https://github.com/digital-asset/daml/pull/17225

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst (hunk 3 above)
Comment (stefanobaghino-da) [reply docs/2.7.0/docs/canton/usermanual/upgrading.rst:225]: [I added it to the deprecated fields](https://github.com/DACH-NY/canton/pull/13207/files#diff-29998a0970c1b9ccd7b6a444b7d0090442702644d090e92b2f13e902054f57b6R84-R89) and that should log a message about this if the field is set explicitly. /cc @thibault-da

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst [hunk 5]
```diff
@@ -180,6 +182,92 @@ Finally, you can ping the participant to see if the system is operational
...
+Unique Contract Keys Deprecation
+""""""""""""""""""""""""""""""""
+The unique-contract-keys parameters for both participant and domain nodes are now marked as deprecated.
+As of this release, the meaning and default value (true) remain unchanged.
+However, contract key uniqueness will not be available in the next major version, featuring multi-domain connectivity.
+If you are already setting this key to false explicitly (preview), this behavior will be the default one after the configuration key is removed.
+If you don't explicitly set this value to false, you are encouraged to evaluate evolving your existing applications and services to avoid relying on this feature.
+You can read more on the topic in the :ref:`documentation <canton_keys>`.
+
+Causality Tracking
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/upgrading.rst:228]:
```suggestion
Causality tracking
```

File: docs/2.7.0/docs/canton/usermanual/upgrading.rst [hunk 6]
```diff
@@ -180,6 +182,92 @@ Finally, you can ping the participant to see if the system is operational
...
+""""""""""""""""""
+An obsolete early access feature to enable causality tracking, related to preview multi-domain, was removed. If you enabled it, you need to remove the following config lines, as they will not compile anymore:
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/upgrading.rst:230]:
```suggestion
An obsolete early access feature to enable causality tracking, related to the preview multi-sync-domain, was removed. If you enabled it, you need to remove the following config lines, as they will not compile anymore:
```

**Review by carrielaben-da — APPROVED** [review]
Some general formatting and grammar changes; otherwise looks good!

**Pull request merged**